GoogleMeetup/src/main/webapp/script.js

```diff
@@ -315,6 +301,168 @@ function initAutocomplete() {
+    }
+
+    maxTravelTime = document.getElementsByClassName('slider')[0].value;
+
+    drawPaths();
+
+    alert("Not Finished Yet!");
+}
+
+
+function validateInput() {
+    if (attendees.length == 0) {
+        alert("You must add at least one person");
+        return false;
+    } if (destinationType == null) {
+        alert("You must select a destination type");
+        return false;
+    }
+    
+    return true;
+}
+
+
+function drawPaths(){
+    var directionsService = new google.maps.DirectionsService();
+
+    for (var i = 0; i < attendees.length; i++) {
+        var request = {
+            origin: attendees[i].address,
+            destination: mapCenter,
+            travelMode: getTransitMode(attendees[i].transport)
+        };
+
+        var directionsRenderer = new google.maps.DirectionsRenderer();
+        directionsRenderer.setMap(map);
+        
+        directionsService.route(request, function(response, status) {
+            if (status === 'OK') {
+                // This is for later when rendering route duration in infowindows
+                // console.log(response.routes[0].legs[0].duration);
+
+                directionsRenderer = new google.maps.DirectionsRenderer();
+                directionsRenderer.setMap(map);
+                directionsRenderer.setDirections(response);
+            }
+        });
+    }
+}
+
+function getTransitMode(transport) {
+    if (transport == "car") {
+        return "DRIVING";
+    } else if (transport == "public") {
+        return "TRANSIT";
+    } else if (transport == "bike") {
+        return "BICYCLING";
+    } else if (transport == "walk") {
+        return "WALKING";
+    }
+}
+
+
+var latLngs = [];
+function getLatLng(attendee) {
+    var address = attendee['address'];
+    geocoder.geocode( { 'address': address}, function(results, status) {
+        if (status == 'OK') {
+            var latLng = results[0].geometry.location;
+            latLngs[latLngs.length] = latLng;
+
+            addAttendeeMarker();
+            centerMap();
+        } else { // For debugging
+            console.log('Geocode was not successful for the following reason: ' + status);
+        }
+    });
+}
+
+
+function centerMap() {
+    if (latLngs.length == 1) {
+        map.setCenter(latLngs[0]);
+        mapCenter = latLngs[0];
+    } else {
+        var midpoint = {
+            "lat":latLngs[0].lat(), 
+            "lng": latLngs[0].lng()
+        };
+
+        for (var i = 1; i < latLngs.length; i++) {
+            midpoint = calculateMidpoint(midpoint.lat, midpoint.lng, latLngs[i].lat(), latLngs[i].lng());
+        }
+
+        map.setCenter(midpoint);
+        mapCenter = midpoint;
+    }
+}
+
+
+function addAttendeeMarker() {
+    var latLng = latLngs[latLngs.length - 1];
+    var marker = new google.maps.Marker({
+        map: map,
+        position: latLng
+    });
+}
+
+
+// Algorithm for geographical midpoint on spherical surface
+function calculateMidpoint (lat1, lng1, lat2, lng2) {
```

Comment:
Add some documentation explaining what this calculation is doing.

Reply:
I've replaced this function with one that averages the latLongs